subprojects/enterprise/src/main/java/org/gradle/internal/tasks/impl/DefaultInputFileProperty.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.tasks.impl;
+
+import org.gradle.internal.tasks.InputFileProperty;
+
+import java.io.File;
+import java.util.Set;
+
+public class DefaultInputFileProperty implements InputFileProperty {
+
+    private final String propertyName;
+    private final Set<File> files;
+
+    public DefaultInputFileProperty(String propertyName, Set<File> files) {
+        this.propertyName = propertyName;
+        this.files = files;
+    }
+
+    @Override
+    public String getPropertyName() {
+        return propertyName;
+    }
+
+    @Override
+    public Set<File> getFiles() {
+        return files;
+    }
```

Comment:
Again, I'm not sure if this should be a `Set` specifically given the cost of constructing such a collection.

Reply:
Should we use `FileCollection`?

Reply:
I would not expose `FileCollection` to `:enterprise` at all. I'd use `Collection<File>`. We can always make it stricter later by changing the return type here to `Set<File>` if this becomes a problem.

Reply:
I think even `Iterable<File>` would be enough for our use case.